libr/cons/cons.c

```diff
@@ -928,14 +928,16 @@ R_API void r_cons_printf_list(const char *format, va_list ap) {
 	va_end (ap3);
 }
 
-R_API void r_cons_printf(const char *format, ...) {
+R_API int r_cons_printf(const char *restrict format, ...) {
```

Comment:
Why do we need `restrict`?

Reply:
I said it already in my second comment on this page:

> the type signature has to be the same like printf (`int printf(const char * restrict format, ...);`).

Reply:
The types have to be compatible because printf gets assigned to it.

Reply:
> I said it already in my second comment on this page:
>>    the type signature has to be the same like printf (int printf(const char * restrict format, ...);).

Sorry, I thought it was just about the `int` return type.
Two questions: 
1) I do see `int printf(const char *format, ...);` in my man page, without the `restrict`.
2) `printf` is assigned to what?

Reply:
The latest standard, C11, declares printf as `int printf(const char * restrict format, ...);`. And C99 too, I guess.

printf is assigned to struct members called `cb_printf`, so they have to be the same type.